### PR TITLE
Feat/edit quiz bank

### PR DESCRIPTION
**Pull request by lilinkw**
- Get quiz data
- Display data  quiz form
- Call Edit form or Add API from quiz form

### CODE REVIEW

File: messages/vi.json [hunk 1]
```diff
@@ -256,12 +256,55 @@
...
+    "signout": "Đăng xuất",
+    "classrooms": {
+      "index": "Lớp học",
+      "new": "Lớp học mới",
+      "my_classrooms": "Lớp học của tôi"
+    },
+    "quizbank": {
+      "index": "Ngân hàng câu hỏi",
+      "new": "Câu hỏi mới",
+      "my_quizbanks": "Ngân hnagf câu hỏi của tôi"
```
Comment (blackmouse572) [inline messages/vi.json:272]: Word

File: messages/en.json [hunk 2]
```diff
@@ -278,9 +278,10 @@
     "support": "Support",
     "signout": "Sign out"
   },
-  "AddQuiz": {
+  "EditQuiz": {
```
Comment (blackmouse572) [inline messages/en.json:281]: Do not delete Addquiz !

File: src/app/[locale]/(main)/quiz/add/components/add-quizbank-form.tsx [hunk 3]
```diff
@@ -74,16 +84,29 @@ const addQuizbankSchema = z.object({
...
 }
-function AddQuizbankForm({ initialValues }: AddQuizbankFormProps) {
+function AddQuizbankForm({
+  initialValues,
+  action = EQuizBankAction.Add,
+  quizBankId
+}: AddQuizbankFormProps) {
   const errori18n = useTranslations("Validations")
-  const i18n = useTranslations("AddQuiz")
+  const i18n = useTranslations("EditQuiz")
```
Comment (blackmouse572) [inline src/app/[locale]/(main)/quiz/add/components/add-quizbank-form.tsx:109]: Add Quiz and Edit quiz have diff range of terms, separate this

File: src/app/[locale]/(main)/quiz/edit/[id]/page.tsx
Comment (blackmouse572) [inline src/app/[locale]/(main)/quiz/edit/[id]/page.tsx:1]: idealy goes for /:id/edit instead of edit/:id based on convention "item/action"

**Review by blackmouse572 — COMMENTED** [review]
Overall, look good. but need to separate the edit and add form into an abstract on. 
You can pass the messages to props or get messages per edit form/ add form. Do not delete i18n namespace in order to maintain

File: src/app/[locale]/(main)/quiz/add/components/add-quizbank-form.tsx (hunk 3 above)
Comment (lilinkw) [reply src/app/[locale]/(main)/quiz/add/components/add-quizbank-form.tsx:109]: But separate Add Quiz and Edit Quiz means that I have to passin NextIntl like this 
-
image
- 
image

Which cause error like this 
image

For solution like this 
image

I can resolve it, but that means I should create 2 form for 2 action. Which is, in my opinion, waste of code
@blackmouse572

Comment (blackmouse572) [reply src/app/[locale]/(main)/quiz/add/components/add-quizbank-form.tsx:109]: Seperate the form with messages props. using mode to pass correct messages

Comment (blackmouse572) [reply src/app/[locale]/(main)/quiz/add/components/add-quizbank-form.tsx:109]: Or using same namespace, with different type

**Review by blackmouse572 — APPROVED** [review]
Look good

**Pull request merged**